detox/src/devices/drivers/android/emulator/EmulatorDriver.js

```diff
@@ -71,12 +71,15 @@ class EmulatorDriver extends AndroidDriver {
   }
 
   async cleanup(deviceId, bundleId) {
-    await this._deviceRegistry.disposeDevice(deviceId);
-    await super.cleanup(deviceId, bundleId);
+    try {
+      await super.cleanup(deviceId, bundleId);
+    } finally {
+      await this._deviceAllocation.deallocateDevice(deviceId)
+    }
   }
 
   async shutdown(deviceId) {
-    await this._deviceAllocation.deallocateDevice(deviceId)
+    await this._emulatorLauncher.shutdown(deviceId);
```

Comment:
The `EmulatorLauncher.shutdown()` executes the shutdown sequence (i.e. via telnet, etc.) instead of `EmulatorDeviceAllocation.deallocateDevice()`.